### ISSUE TITLE
full_transcribe is much slower on Android Devices 

### ISSUE THREAD

**Issue by visla-yewang** [ggml-org/whisper.cpp#959]
I am going to implement one app in Android just like continuous transcribe audio from Mic, which is just like the stream.cpp implementation. 

but the issue is when run on Android, I found that , 3 seconds audio input need 3 seconds for transcribe, which is not acceptable for the real time transcribe. 

is there any reason why it is so slow on Android? I also tested on IOS, it is much faster than Android Platform.

**Comment by visla-yewang** [comment]
I am using ggml-tiny.en.bin model, and the device hardware spec is XIaoMi Mi 11,  8 GB Memory

the CPU Info is  as below.

processor	: 0
BogoMIPS	: 38.40
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm lrcpc dcpop asimddp
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x2
CPU part	: 0xd05
CPU revision	: 0

processor	: 1
BogoMIPS	: 38.40
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm lrcpc dcpop asimddp
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x2
CPU part	: 0xd05
CPU revision	: 0

processor	: 2
BogoMIPS	: 38.40
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm lrcpc dcpop asimddp
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x2
CPU part	: 0xd05
CPU revision	: 0

processor	: 3
BogoMIPS	: 38.40
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm lrcpc dcpop asimddp
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x2
CPU part	: 0xd05
CPU revision	: 0

processor	: 4
BogoMIPS	: 38.40
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm lrcpc dcpop asimddp
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd41
CPU revision	: 0

processor	: 5
BogoMIPS	: 38.40
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm lrcpc dcpop asimddp
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd41
CPU revision	: 0

processor	: 6
BogoMIPS	: 38.40
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm lrcpc dcpop asimddp
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd41
CPU revision	: 0

processor	: 7
BogoMIPS	: 38.40
Features	: fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm lrcpc dcpop asimddp
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x1
CPU part	: 0xd44
CPU revision	: 0

Hardware	: Venus based on Qualcomm Technologies, Inc SM8350

**Comment by visla-yewang** [comment]
I also added the operation time log in the demo android app, the  code snippet is as below.
    LOGI("About to run whisper_full");
    struct timeval tv;
    gettimeofday(&tv, NULL);
    LOGI("%ld\n", tv.tv_sec * 1000 + tv.tv_usec / 1000);
    if (whisper_full(context, params, audio_data_arr, audio_data_length) != 0) {
        LOGI("Failed to run the model");
    } else {
        whisper_print_timings(context);
    }
    struct timeval tv1;
    gettimeofday(&tv1, NULL);
    LOGI("%ld\n", tv1.tv_sec * 1000 + tv1.tv_usec / 1000);

the output of the execute is as below.

2023-05-24 11:03:55.631  6697-6773  JNI                     pid-6697                             I  1684897435631
2023-05-24 11:03:58.761  6697-6773  JNI                     pid-6697                             I  1684897438761

the time difference is 3130 ms.

the input voice is about 3 seconds

**Comment by Martin-Laclaustra** [comment]
Make sure that you test a "release" APK and not a "debug" version.
Apparently this makes an important difference https://github.com/ggerganov/whisper.cpp/issues/518#issuecomment-1537294226

**Comment by visla-yewang** [comment]
of course, I am using an release APK, not a debug one, for sure, but the still it take times

**Comment by haydenkaizeta** [comment]
Mine is even slower, Pixel 4XL, 3 second recording takes about 30 seconds with fullTranscribe. tiny.en model, no other changes to the project

**Comment by wuchuifeng** [comment]
> I am going to implement one app in Android just like continuous transcribe audio from Mic, which is just like the stream.cpp implementation.
> 
> but the issue is when run on Android, I found that , 3 seconds audio input need 3 seconds for transcribe, which is not acceptable for the real time transcribe.
> 
> is there any reason why it is so slow on Android? I also tested on IOS, it is much faster than Android Platform.

你好，我最近也在找关于用Whisper.cpp做android离线实时语音识别的demo，你的问题解决了吗，可以有偿跟你咨询，我的联系方式：13673668675
